compose/foundation/foundation/src/uikitMain/kotlin/androidx/compose/foundation/text/selection/TextFieldSelectionManager.uikit.kt

```diff
@@ -136,3 +136,45 @@ private fun calculateSelectionMagnifierCenterIOS(
 }
 
 private const val HideThresholdDp = 36
+
+/**
+ * Multiplier for height tolerance calculation.
+ * iOS has different location for the knobs of selection handles (leading - top, trailing - bottom),
+ * and they might be not seen in mixed BiDi text, so this tolerance is required for their visibility in this case
+ * if more than 80% of selection handle is visible, it should be shown
+ */
+private const val HeightToleranceFactor = 0.2f
+
+/**
+ * Whether the selection handle is in the visible bound of the TextField.
+ */
+internal actual fun TextFieldSelectionManager.isSelectionHandleInVisibleBound(
+    isStartHandle: Boolean
+): Boolean {
+    fun getSelectionHandleVisibilityPointPosition(isStartHandle: Boolean): Offset {
+        val textLayoutResult = state?.layoutResult?.value ?: return Offset.Unspecified
+
+        // If layout and value are out of sync, return unspecified.
+        // This will be called again once they are in sync.
+        val transformedText = transformedText ?: return Offset.Unspecified
+        val layoutInputText = textLayoutResult.layoutInput.text.text
+        if (transformedText.text != layoutInputText) return Offset.Unspecified
+
+        val offset = if (isStartHandle) value.selection.start else value.selection.end
+        val line = textLayoutResult.getLineForOffset(offset)
+        // This happens if maxLines is set but the offset is on a line >= maxLines.
+        if (line >= textLayoutResult.lineCount) return Offset.Unspecified
+
+        val x = textLayoutResult.getHorizontalPosition(offset, isStartHandle, value.selection.reversed)
+
+        val heightTolerance = textLayoutResult.multiParagraph.getLineHeight(line) * HeightToleranceFactor
+        val y = if (isStartHandle) textLayoutResult.getLineTop(line) + heightTolerance else textLayoutResult.getLineBottom(line) - heightTolerance
+
+        return Offset(x, y)
+    }
+
+    val visibleBounds = state?.layoutCoordinates?.visibleBounds()
+    val handlePosition = getSelectionHandleVisibilityPointPosition(isStartHandle)
+
+    return visibleBounds?.containsInclusive(handlePosition) ?: false
```

Comment:
It seems alternative of `containsInclusive` with `HeightToleranceFactor` should be defined instead of reinventing `getHandlePosition` with their corner cases and transformations

Reply:
I rewrote it like this because I wanted to make proper check for coordinates of left and right selection handles (line 171), but it seems like it's not worth it.